arch/xtensa/core/xtensa_mmu.c

```diff
@@ -329,10 +339,33 @@ void z_xtensa_mmu_smp_init(void)
 	xtensa_mmu_init(false);
 }
 
+#ifdef CONFIG_ARCH_HAS_RESERVED_PAGE_FRAMES
+/* Zephyr's linker scripts for Xtensa usually puts
+ * something before z_mapped_start (aka .text),
+ * i.e. vecbase, so that we need to reserve those
+ * space or else k_mem_map() would be mapping those,
+ * resulting in faults.
+ */
```

Comment:
Mentioned this in the other PR, but I really think this particular code belongs in the SOC layer, as that's where memory map responsibility is and we can be a TON more prescriptive (e.g. "reserve the 4k vector region area at the start of hpsram" or whatever instead of "linker scripts usually...").